CMakeLists.txt

```diff
@@ -0,0 +1,295 @@
+cmake_minimum_required(VERSION 3.7 FATAL_ERROR)
+
+cmake_policy(SET CMP0048 NEW)
+cmake_policy(SET CMP0026 OLD) # nix_install
+
+project(nix VERSION 1.11.11)
+message(STATUS "Project: ${PROJECT_NAME}, version: ${PROJECT_VERSION}")
+enable_testing()
+
+if(NOT CMAKE_BUILD_TYPE)
+  set(CMAKE_BUILD_TYPE "Release" CACHE STRING
+      "Choose the type of build, options are: Debug Release RelWithDebInfo MinSizeRel." FORCE)
+endif(NOT CMAKE_BUILD_TYPE)
+message(STATUS "Build type: ${CMAKE_BUILD_TYPE}")
+
+# Construct a Nix system name (like "i686-linux").
+set(SYSTEM "${CMAKE_SYSTEM_PROCESSOR}-${CMAKE_SYSTEM_NAME}")
+
+# State should be stored in /nix/var, unless the user overrides it explicitly.
+set(CMAKE_INSTALL_LOCALSTATEDIR "/nix/var" CACHE PATH "")
+
+# variables are prepended (if necessary) by the value of CMAKE_INSTALL_PREFIX.
+include(GNUInstallDirs)
+
+set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${CMAKE_SOURCE_DIR}/cmake)
+include(NixCommon)
+include(NixInstall)
+
+include(FindPkgConfig)
+include(CheckIncludeFiles)
+include(CheckIncludeFileCXX)
+include(CheckFunctionExists)
+include(CheckSymbolExists)
+include(CheckLibraryExists)
+#include(CheckPrototypeExists)
+include(CheckStructHasMember)
+
+include(CMakePackageConfigHelpers)
+
+# TODO: Use 64-bit file system calls so that we can support files > 2 GiB.
+# AC_SYS_LARGEFILE
+
+# Check for pubsetbuf.
+check_cxx_source_compiles("
+    #include <iostream>
+    using namespace std;
+    static char buf[1024];
+    int main()
+    {
+        cerr.rdbuf()->pubsetbuf(buf, sizeof(buf));
+    }"
+    HAVE_PUBSETBUF
+)
+
+check_function_exists(statvfs HAVE_STATVFS)
+check_function_exists(pipe2 HAVE_PIPE2)
+
+# Check for lutimes, optionally used for changing the mtime of symlinks.
+check_function_exists(lutimes HAVE_LUTIMES)
+
+# Check whether the store optimiser can optimise symlinks.
```

Comment:
We can remove the configuration tests for the following variables, which are not checked in the source code itself: 

- HAVE_BZLIB_H
- HAVE_BZLIB (so we remove the check for BZ2_bzWriteOpen)
- HAVE_DIRENT_H
- HAVE_INTTYPES_H
- HAVE_MEMORY_H
- HAVE_NDIR_H
- HAVE_STDINT_H
- HAVE_STDLIB_H
- HAVE_STRINGS_H
- HAVE_STRING_H
- HAVE_SYS_DIR_H
- HAVE_SYS_NDIR_H
- HAVE_SYS_STAT_H
- HAVE_SYS_TYPES_H
- HAVE_UNISTD_H
- STDC_HEADERS
- HAVE_SETRESUID (these last two are checked for alongside of lchown)
- HAVE_SETREUID

Note: I'm not sure which autotools macro (i.e. which line in configure.ac) is responsible for generating most of these (but it should be removed).